Devs-AA/Barefoot
# Ft book accommodation 168662118

#### What does this PR do?
Enable requesters book accommodation for a trip

#### Description of Task to be completed?
  - Enable requesters to book an accommodation
  - Ensure only requesters with approved requests can book an accommodation

#### How should this be manually tested?
- git clone this repo using the command `git@github.com:Devs-AA/Barefoot.git`
  - cd to the project
  - checkout to the branch using the command `git checkout ft-book-accommodation-168662118`
  - create a `.env` file and set PORT,JWT_SECRET and DB variables as specified in the .env.example file.
  - Run the command `npm install` in the terminal.
  - Open a terminal and run `npm test`.
  - Open a terminal and run `npm run migrate:dev` and `npm run seed:dev`
  1. Open Postman, make a post request to `localhost:<port>/api/v1/users/auth/login`, with the below in the request body.
- `{email: requester2@gmail.com, password: Password1$}`
2. Copy the token in the reponse
3. Make a POST request to `http:localhost:<port>/api/v1/requests` with the token obtained above in the request header as `authorization`=`Bearer <token obtained>`
 4. Send a request object containing the following for a one-way trip:
         `{
           reason: 'Business',
           departmentId: 2,
           tripType: 'oneWay',
        }`

5. Make a post request to `localhost:<port>/api/v1/users/auth/login`, with the below  in the request body.
 `{email: barefoot2@gmail.com, password: Password1$}`
6. Make a PATCH request to `http:localhost:<port>/api/v1/requests<requestId>` with the request body containing. Note that status can only be approved or rejected
          `{
            "status": "approved"
           }`
 7. Make a POST request to `http://localhost:<port>/api/v1/accommodations/<accommodationId>`, with the same authorization header used in creating request. The request body should comply with that in the image below
![Screenshot (103)](https://user-images.githubusercontent.com/46751363/69768451-e5f8df00-1180-11ea-83e9-d8efe5f3708f.png)
 
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168662118](https://www.pivotaltracker.com/story/show/168662118)
#### Screenshots (if appropriate)
![Screenshot (102)](https://user-images.githubusercontent.com/46751363/69731095-8f13eb00-1129-11ea-9a74-f17541b7f268.png)



#### Questions
N/A
